app/services/users/import_data/points.rb

```
@@ -1,96 +1,163 @@
+    begin
+      result = Point.upsert_all(
+        normalized_batch,
+        unique_by: %i[lonlat timestamp user_id],
+        returning: %w[id],
+        on_duplicate: :skip
+      )
+
+      batch_created = result&.count.to_i
+      @total_created += batch_created
+
+      logger.debug "Processed batch of #{@buffer.size} points, created #{batch_created}, total created: #{@total_created}"
```

RuboCop (1.81.1)

[convention] Line is too long. [122/120]

(Layout/LineLength)